### PR TITLE
V14: Cache user by id

### PR DESCRIPTION
**Pull request by Zeegaan**
# Notes
- We need to cache user by id aswell, because all content have creatorId (as int) and thus when we index content, user by id gets called for each content item, therefore we need to cache the user to avoid a lot of db calls.

# How to test
- Put a breakpoint in the `UserRepository.Get(int id)` method
- Assert that first time a user gets retrieved, the `GetCacheItem` should return null (not in the cache
- Assert that the second time the user gets retrieved `GetCacheItem` should now have a user.
- Update the users name (or something else, name is easiest)
- Get the user again, and assert that the `GetCacheItem` returns null (Updated user should NOT be in the cache)
- Get the user once last time, and assert that it is indeed cached again, with the updated name

### CODE REVIEW

File: src/Umbraco.Core/Cache/Refreshers/Implement/UserCacheRefresher.cs [hunk 1]
```diff
@@ -45,5 +45,18 @@ public override void Refresh(Guid id)
...
 
+    public override void Refresh(int id)
+    {
+        Remove(id);
+        base.Refresh(id);
+    }
+
+    public override void Remove(int id)
+    {
+        string cacheKey = $"uRepo_{typeof(IUser).Name}_" + id;
```
Comment (Nuklon) [inline src/Umbraco.Core/Cache/Refreshers/Implement/UserCacheRefresher.cs:56]: Use RepositoryCacheKeys or CacheKeys here to construct this? Same below.

Comment (Zeegaan) [reply src/Umbraco.Core/Cache/Refreshers/Implement/UserCacheRefresher.cs]: Good point, done 👍 😁

File: src/Umbraco.Core/Cache/Refreshers/Implement/UserCacheRefresher.cs [hunk 2]
```diff
@@ -2,60 +2,58 @@
...
     public override string Name => "User Cache Refresher";
 
-    #endregion
+    public record JsonPayload
+    {
+        public int Id { get; init; }
 
-    #region Refresher
+        public Guid Key { get; init; }
+    }
```
Comment (Nuklon) [inline src/Umbraco.Core/Cache/Refreshers/Implement/UserCacheRefresher.cs:27]: This creates a lot of bulk, can also be a class.

Comment (Zeegaan) [reply src/Umbraco.Core/Cache/Refreshers/Implement/UserCacheRefresher.cs:27]: This is the general pattern used for these JsonPayloads, see `ContentCacheRefresher` & `MemberCacheRefresher` 😁 

And IMO records are the correct type here (also the plan to change the others) as this is used purely to pass data 😊

**Review by bergmania — APPROVED** [review]

File: src/Umbraco.Core/Cache/Refreshers/Implement/UserCacheRefresher.cs (hunk 2 above)
Comment (Nuklon) [reply src/Umbraco.Core/Cache/Refreshers/Implement/UserCacheRefresher.cs:27]: The compiler creates a lot of unnecessary code for records, these simple types are better suited for a regular class.

**Pull request merged**